src/main/com/cqfn/degitx/simulator/Router.kt

```diff
@@ -0,0 +1,22 @@
+package com.cqfn.degitx.simulator
+
+import java.util.*
+
+/**
+ * Smart Router representation.
+ * As part of global networks connected with other Servers via Edges.
+ *
+ * @see Graph
+ * @see Edge
+ */
+class Router(override var state: State,
```

Comment:
@IlyaMoskva just an idea: maybe we can make a router as one implementation of node software? The logic could be: it accepts requests from connected nodes via queue and sends to local target or proxies to next level router

Reply:
@g4s8, Yes I followed this idea here. Both classes, Node and Router are implementations of Server interface. Server is a physical device, and Router is a "corner case" Node without storage and Node software. It has the same method to redirect request, but can't be an addressee.
Maybe it will become redundant later, for now I would keep it separately.